griptape/tasks/base_text_input_task.py

```diff
@@ -24,7 +25,7 @@ class BaseTextInputTask(RuleMixin, BaseTask, ABC):
         alias="input",
     )
 
-    @property
+    @cached_property
```

Comment:
is there concern here with having this cached but having something like `self.full_context` change? those changes wont get picked up

Reply:
I have this question too. What is expected in a situation like this? (excuse any usage mistakes, focus on the general point of mutating full_context in between input() calls):

```python
t = Task("foo: {{ foo }}")
t.full_context["foo"] = "foo"
x = t.input()

assert x == 'foo: foo'

t.full_context["foo"] = "bar"
y = t.input()

# after mutating full_context, should we expect this?
assert y == 'foo: foo'
# or this?
assert y == 'foo: bar' 
```

Reply:
yep exactly. the property is cached so its always going to print out "foo: foo"

Reply:
Ah I see. Yes, it will print out `foo: foo`. The cache does get invalidated when Task is reset, but I can see this leading to some weird and difficult bugs. Such as:

```python
def test_input_context(self):
        pipeline = Pipeline(
            tasks=[
                PromptTask(
                    "foo",
                    prompt_driver=MockPromptDriver(),
                    on_before_run=lambda task: print(task.children[0].input),
                ),
                PromptTask("{{ parent_output }}", prompt_driver=MockPromptDriver()),
            ]
        )

        pipeline.run()

        assert pipeline.tasks[1].input.value == "mock output 1"
```

Lemme see if I can revert and still get the behavior I'm looking to achieve.

Reply:
Reverted and we're good 👍